### PR TITLE
fix(hook): Add missing hook_table_revert impl

### PR DESCRIPTION
**Pull request by icex2**
fix(hook): Add missing hook_table_revert impl

Allow hooks to cleanup when they are shut down.

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/djhackersdev/bemanitools/pull/286).
* #298
* #297
* #296
* #295
* #294
* #293
* #292
* #291
* #290
* #289
* #288
* #287
* __->__ #286
* #285

### CODE REVIEW

File: src/main/hook/table.c [hunk 1]
```diff
@@ -101,6 +156,33 @@ static void hook_table_apply_to_iid(
...
+    const struct hook_symbol *sym;
+
+    i = 0;
+
+    while (pe_iid_get_iat_entry(target, iid, i++, &iate) == S_OK) {
+        for (j = 0; j < nsyms; j++) {
+            sym = &syms[j];
+
+            if (hook_table_match_proc(&iate, sym)) {
+                // Only revert-able if the original pointer was stored previously
```
Comment (willxinc) [inline src/main/hook/table.c:177]: should we log in this case?

**Review by willxinc — APPROVED** [review]
Looks good to me, just a question about the failure case, as well as an overall question about when this would be used

File: src/main/hook/table.c (hunk 1 above)
Comment (icex2) [reply src/main/hook/table.c:177]: We don't have logging available there. This is the capnhook package. =/

**Comment by icex2** [comment]
> Looks good to me, just a question about the failure case, as well as an overall question about when this would be used

Added reasoning to commit body.

**Pull request merged**